### PR TITLE
Button cleanup

### PR DESCRIPTION
**Pull request by nlsvgtr**
# Description

Link buttons do not show correctly; fix this and cleanup the resulting code

## Type of change

Code improvement

### CODE REVIEW

File: src/button/component/button.jsx [hunk 1]
```diff
@@ -56,18 +53,15 @@ export default class OpenStadComponentButton extends OpenStadComponent {
       labelHTML = <div className="osc-button-label"><div className="osc-elipsis">{self.config.label}</div></div>
     }
 
+    let onClick = typeof self.props.onClick != 'undefined' ? self.props.onClick : self.state.onClick || null;
```
Comment (rudivanhierden) [inline src/button/component/button.jsx:56]: `onClick` doesn't seem to exist in the state, so I think we can remove it here?

```suggestion
    let onClick = typeof self.props.onClick != 'undefined' ? self.props.onClick : null;
```

**Review by rudivanhierden — COMMENTED** [review]
Since we are moving from an `a` element to a `div` with an `onClick`, it might be good to check for accessibility as well. Things like `aria-role="link"` and a `tabindex="0"`. 

We can create an issue for it and work it out later as well (maybe throughout all components?), but it's good to keep in mind.

**Review by rudivanhierden — APPROVED** [review]
Thanks, LGTM! 💪

**Comment by nlsvgtr** [comment]
Both, I think. I created the issue and updated the code. Thanks.

**Pull request merged**